File: backend/app/live_data/schemas/market.py
"""Normalized Mandi and commodity market schemas."""

from datetime import date
from enum import Enum
from pydantic import Field

from app.live_data.schemas.common import BaseLiveDataResponse


class MandiTrend(str, Enum):
    """Commodity price trend."""

    RISING = "RISING"
    FALLING = "FALLING"
    STABLE = "STABLE"
    UNKNOWN = "UNKNOWN"


class MarketPriceObservation(BaseLiveDataResponse):
    """Normalized commodity mandi price observation."""

    commodity: str
    variety: str = "Common"
    market: str
    district: str
    state: str
    arrival_date: date
    min_price_inr_quintal: float
    max_price_inr_quintal: float
    modal_price_inr_quintal: float
    msp_inr_quintal: float | None = None
    price_trend: MandiTrend = MandiTrend.STABLE
    arrivals_tonnes: float | None = None
    currency: str = "INR"
    unit: str = "Quintal"

File: frontend/src/types/market.ts
/**
 * TypeScript types for Mandi Market Prices matching backend live_data schemas.
 *
 * See: backend/app/live_data/schemas/market.py
 */

export type MandiTrend = 'RISING' | 'FALLING' | 'STABLE' | 'UNKNOWN';

export interface MarketPriceObservation {
  commodity: string;
  variety: string;
  market: string;
  district: string;
  state: string;
  arrival_date: string;
  min_price_inr_quintal: number;
  max_price_inr_quintal: number;
  modal_price_inr_quintal: number;
  msp_inr_quintal?: number | null;
  price_trend: MandiTrend;
  arrivals_tonnes?: number | null;
  currency: string;
  unit: string;
  source_dataset?: string;
  fetched_at?: string;
  freshness_level?: 'FRESH' | 'RECENT' | 'STALE' | 'EXPIRED' | 'UNKNOWN';
}
